Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -332,34 +334,54 @@ public void FailPlugin(Assembly assembly)
             ChangePluginState(plugin, PluginStatus.Malfunctioned);
         }
 
-        /// <summary>
-        /// Saves the manifest back to disk.
-        /// </summary>
-        /// <param name="manifest">The <see cref="PluginManifest"/> to save.</param>
-        /// <param name="path">The path where to save the manifest.</param>
-        /// <returns>True if successful.</returns>
+        /// <inheritdoc/>
         public bool SaveManifest(PluginManifest manifest, string path)
         {
-            if (manifest == null)
-            {
-                return false;
-            }
-
             try
             {
                 var data = JsonSerializer.Serialize(manifest, _jsonOptions);
                 File.WriteAllText(Path.Combine(path, "meta.json"), data);
                 return true;
             }
-#pragma warning disable CA1031 // Do not catch general exception types
-            catch (Exception e)
-#pragma warning restore CA1031 // Do not catch general exception types
+            catch (ArgumentException e)
             {
-                _logger.LogWarning(e, "Unable to save plugin manifest. {Path}", path);
+                _logger.LogWarning(e, "Unable to save plugin manifest due to invalid value. {Path}", path);
                 return false;
             }
         }
 
+        /// <inheritdoc/>
+        public bool GenerateManifest(PackageInfo packageInfo, Version version, string path)
+        {
+            var versionInfo = packageInfo.Versions.First(v => v.Version == version.ToString());
+            var url = packageInfo.ImageUrl ?? string.Empty;
+            var imageFilename = url.Substring(url.LastIndexOf('/') + 1, url.Length);
+
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(url, Path.Join(path, imageFilename));
+            }
+
+            var manifest = new PluginManifest
+            {
+                Category = packageInfo.Category,
+                Changelog = versionInfo.Changelog ?? string.Empty,
+                Description = packageInfo.Description,
+                Id = new Guid(packageInfo.Id),
+                Name = packageInfo.Name,
+                Overview = packageInfo.Overview,
+                Owner = packageInfo.Owner,
+                TargetAbi = versionInfo.TargetAbi ?? string.Empty,
+                Timestamp = DateTime.Parse(versionInfo.Timestamp ?? string.Empty),
```

Comment:
```suggestion
                Timestamp = string.IsNullOrEmpty(versionInfo.Timestamp) ? DateTime.MinValue : DateTime.Parse(versionInfo.Timestamp)
```

`DateTime.Parse` can't handle `string.Empty`